tests/test_data_types.py

```diff
@@ -155,3 +183,126 @@ def test_category_deserialized_value_is_not_in_choices():
     cat = category(choices=['one', 'two', 'three'])
     with pytest.raises(InvalidInputError):
         cat.deserialize('four')
+
+# FILE -------------------------------------------------------------------------
+def test_file_to_dict():
+    f = file()
+    obj = f.to_dict()
+    assert obj['name'] == 'file'
+    assert obj['type'] == 'file'
+
+def test_file_to_dict_folder():
+    f = file(is_folder=True)
+    obj = f.to_dict()
+    assert obj['name'] == 'file'
+    assert obj['type'] == 'file'
+    assert obj['isFolder'] == True
+
+def test_file_serialization_base():
+    f = file()
+    assert 'file.txt' == f.serialize('file.txt')
+
+def test_file_serialization_relative():
+    f = file()
+    assert 'folder/file.txt' == f.serialize('folder/file.txt')
+
+def test_file_serialization_absolute():
+    f = file()
+    assert '/home/user/file.txt' == f.serialize('/home/user/file.txt')
+
+def test_file_serialization_remote():
+    f = file()
+    url = 'https://github.com/runwayml/model-sdk/archive/0.0.57.tar.gz'
+    assert url == f.serialize(url)
+## TODO: accept ftp:// protocol
+#     url = 'ftp://demo:password@test.rebex.net/readme.txt'
+#     assert url == f.serialize(url)
+
+def test_file_serialization_base_folder():
+    f = file(is_folder=True)
+    assert 'folder' == f.serialize('folder')
+
+def test_file_serialization_relative_folder():
+    f = file(is_folder=True)
+    assert 'folder/folder' == f.serialize('folder/folder')
+
+def test_file_serialization_absolute_folder():
+    f = file(is_folder=True)
+    assert '/home/user/folder' == f.serialize('/home/user/folder')
+
+def test_file_serialization_remote_folder():
+    f = file(is_folder=True)
+    url = 'https://github.com/runwayml/model-sdk/archive/0.0.57.tar.gz'
+    assert url == f.serialize(url)
+## TODO: accept ftp:// protocol
+#     url = 'ftp://demo:password@test.rebex.net/'
+#     assert url == f.serialize(url)
+
+def test_file_deserialization_base():
+    f = file()
+    assert 'file.txt' == f.deserialize('file.txt')
+
+def test_file_deserialization_relative():
+    f = file()
+    assert 'folder/file.txt' == f.deserialize('folder/file.txt')
+
+def test_file_deserialization_absolute():
+    f = file()
+    assert '/home/user/file.txt' == f.deserialize('/home/user/file.txt')
+
+def test_file_deserialization_remote():
+    f = file()
+    url = 'https://github.com/runwayml/model-sdk/archive/0.0.57.tar.gz'
+    path = f.deserialize(url)
+    assert os.path.exists(path)
```

Comment:
I would add a check here for checking that the file/folder at the given path has the content expected, e.g. a file size check

Reply:
Good call! Added this type of check with `check_expected_contents_for_0057_tar_download()`